addons/sale/models/sale.py

```diff
@@ -395,7 +395,7 @@ def onchange_partner_id(self):
             'partner_invoice_id': addr['invoice'],
             'partner_shipping_id': addr['delivery'],
         }
-        user_id = partner_user.id
+        user_id = self.env.context.get('sale_force_user_id') or partner_user.id
```

Comment:
Hello @tde-banana-odoo 

This fix is a bit strange.
It looks like we are dirty-fixing the fact that the sale.order model does not correctly support default values by using a context key.

Couldn't we rather adapt the method to correctly support default values?

Something like:

```py
user_id = self.env.context.get('default_user_id') or partner_user.id
if not user_id and not self.env.context.get('not_self_saleperson'):
    user_id =  self.env.uid
if user_id and self.user_id.id != user_id:
    values['user_id'] = user_id
```

Or is that to avoid side-effects on "old" stable?

(@dht-odoo don't change anything yet, I'm just making an example here)

Reply:
It is not really a default, it is a forced value coming from outside and that has precedence on other values. This is therefore not really like a default for me. Using a dedicated context key makes this more explicit.

Indeed the whole user / team computation on SO could be cleaned (notably after converting into compute in master), but this fix has the advantage of being localized, explicit and not changing current behavior of default context key.

Your move :D .

Reply:
Mmmh.

Still not a fan, that basically means you can't set a default_partner_id and a default_user_id in the same context for a sale.order, as it will almost never work (the user will be overridden).

But I guess the potential side effects are too big on stable, maybe we should keep the localized context key fix and rework in master.

Reply:
Indeed, I wouldn't change anything in stable in sale if possible.